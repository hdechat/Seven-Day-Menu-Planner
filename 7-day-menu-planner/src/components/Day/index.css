.day {
  width: 10rem;
}

h6 {
  margin: 1.5rem auto 0rem auto;
}