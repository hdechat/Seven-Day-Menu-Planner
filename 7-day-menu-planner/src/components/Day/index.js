import React from 'react';
import RecipeCard from '../RecipeCard';

const Day = ({ menu }) => {
  return (
    <section>
      <h6>Breakfast</h6>
      <RecipeCard recipe={menu.breakfast} displayAddToMenu={false} />
      <h6>Lunch</h6>
         <RecipeCard recipe={menu.lunch} displayAddToMenu={false} />
      <h6>Dinner</h6>
         <RecipeCard recipe={menu.dinner} displayAddToMenu={false} />
    </section>
  )
}

export default Day;
