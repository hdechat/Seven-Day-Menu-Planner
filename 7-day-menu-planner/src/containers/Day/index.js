import { connect } from 'react-redux';
import { withRouter } from 'react-router';
import Day from '../../components/Day';
import { removeFromMenu } from '../../actions';

export const mapStateToProps = state => ({
  menu: state.menu
});

export const mapDispatchToProps = dispatch => ({
  removeFromMenu: (weekday, mealTime) => dispatch(removeFromMenu(weekday, mealTime))
});

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(Day));
