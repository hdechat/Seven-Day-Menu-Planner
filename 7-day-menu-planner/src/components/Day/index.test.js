import React from 'react';
import Day from './index.js';
import { shallow } from 'enzyme';

describe('Day', () => {
  let wrapper;
  let mockMenu;
  let mockRemoveFromMenu;
  let mockRemoveFromGroceryList;
  let mockRemoveMenuItemFromStorage;
  let mockRemoveGroceryListFromStorage;

  beforeEach(() => {
    mockRemoveFromMenu = jest.fn();
    mockRemoveFromGroceryList = jest.fn();
    mockRemoveGroceryListFromStorage = jest.fn();
    mockRemoveMenuItemFromStorage = jest.fn();

    mockMenu = {
      sunday: {
        breakfast: {title: 'Recipe Name', ingredients: ['apples']},
        lunch: {},
        dinner: {}
      }
    };
    wrapper = shallow(<Day 
      day='sunday' 
      removeFromGroceryList={mockRemoveFromGroceryList} 
      removeFromMenu={mockRemoveFromMenu} 
      menu={mockMenu}
      removeGroceryListFromStorage={mockRemoveGroceryListFromStorage}
      removeMenuItemFromStorage={mockRemoveMenuItemFromStorage}/>);
  });

  it('matches snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('calls removeFromMenu on button click', () => {
    wrapper.find('button').at(0).simulate('click')

    expect(mockRemoveFromMenu).toHaveBeenCalledWith('sunday', 'breakfast');
    expect(mockRemoveFromGroceryList).toHaveBeenCalledWith(['apples']);
    expect(mockRemoveMenuItemFromStorage).
    toHaveBeenCalledWith('sunday', 'breakfast',  
      {ingredients: ['apples'], title: 'Recipe Name'});
    expect(mockRemoveGroceryListFromStorage).toHaveBeenCalledWith(['apples']);
  });
});
