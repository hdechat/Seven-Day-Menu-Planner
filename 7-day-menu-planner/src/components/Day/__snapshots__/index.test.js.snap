// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Day matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Day
    day="sunday"
    menu={
        Object {
            "sunday": Object {
              "breakfast": Object {
                "ingredients": Array [
                  "apples",
                ],
                "title": "Recipe Name",
              },
              "dinner": Object {},
              "lunch": Object {},
            },
          }
    }
    removeFromGroceryList={[Function]}
    removeFromMenu={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h6>
          Breakfast
</h6>,
        <button
          className="display"
          onClick={[Function]}
>
          Remove Recipe
</button>,
        <Connect(Component)
          displayAddToMenu={false}
          displayLinkText={false}
          recipe={
                    Object {
                              "ingredients": Array [
                                "apples",
                              ],
                              "title": "Recipe Name",
                            }
          }
/>,
        <h6>
          Lunch
</h6>,
        <button
          className="no-display"
          onClick={[Function]}
>
          Remove Recipe
</button>,
        <Connect(Component)
          displayAddToMenu={false}
          displayLinkText={false}
          recipe={Object {}}
/>,
        <h6>
          Dinner
</h6>,
        <button
          className="no-display"
          onClick={[Function]}
>
          Remove Recipe
</button>,
        <Connect(Component)
          displayAddToMenu={false}
          displayLinkText={false}
          recipe={Object {}}
/>,
      ],
      "className": "day",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Breakfast",
        },
        "ref": null,
        "rendered": "Breakfast",
        "type": "h6",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Remove Recipe",
          "className": "display",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Remove Recipe",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "displayAddToMenu": false,
          "displayLinkText": false,
          "recipe": Object {
            "ingredients": Array [
              "apples",
            ],
            "title": "Recipe Name",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Lunch",
        },
        "ref": null,
        "rendered": "Lunch",
        "type": "h6",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Remove Recipe",
          "className": "no-display",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Remove Recipe",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "displayAddToMenu": false,
          "displayLinkText": false,
          "recipe": Object {},
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Dinner",
        },
        "ref": null,
        "rendered": "Dinner",
        "type": "h6",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Remove Recipe",
          "className": "no-display",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Remove Recipe",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "displayAddToMenu": false,
          "displayLinkText": false,
          "recipe": Object {},
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h6>
            Breakfast
</h6>,
          <button
            className="display"
            onClick={[Function]}
>
            Remove Recipe
</button>,
          <Connect(Component)
            displayAddToMenu={false}
            displayLinkText={false}
            recipe={
                        Object {
                                    "ingredients": Array [
                                      "apples",
                                    ],
                                    "title": "Recipe Name",
                                  }
            }
/>,
          <h6>
            Lunch
</h6>,
          <button
            className="no-display"
            onClick={[Function]}
>
            Remove Recipe
</button>,
          <Connect(Component)
            displayAddToMenu={false}
            displayLinkText={false}
            recipe={Object {}}
/>,
          <h6>
            Dinner
</h6>,
          <button
            className="no-display"
            onClick={[Function]}
>
            Remove Recipe
</button>,
          <Connect(Component)
            displayAddToMenu={false}
            displayLinkText={false}
            recipe={Object {}}
/>,
        ],
        "className": "day",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Breakfast",
          },
          "ref": null,
          "rendered": "Breakfast",
          "type": "h6",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Remove Recipe",
            "className": "display",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Remove Recipe",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "displayAddToMenu": false,
            "displayLinkText": false,
            "recipe": Object {
              "ingredients": Array [
                "apples",
              ],
              "title": "Recipe Name",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Lunch",
          },
          "ref": null,
          "rendered": "Lunch",
          "type": "h6",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Remove Recipe",
            "className": "no-display",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Remove Recipe",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "displayAddToMenu": false,
            "displayLinkText": false,
            "recipe": Object {},
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Dinner",
          },
          "ref": null,
          "rendered": "Dinner",
          "type": "h6",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Remove Recipe",
            "className": "no-display",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Remove Recipe",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "displayAddToMenu": false,
            "displayLinkText": false,
            "recipe": Object {},
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
